### PR TITLE
regression analysis functions and natural log function

### PR DESCRIPTION
**Pull request by PeterAustinMoore**
Added functions and docs for:
`regr_slope`
`regr_intercept`
`regr_r2`
`ln`

### CODE REVIEW

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 1]
```diff
@@ -282,6 +282,15 @@ object SoQLFunctions {
   val MedianDisc = f("median_disc", FunctionName("median"), Map("a" -> (Ordered -- NumLike)), Seq(VariableType("a")), Seq.empty, VariableType("a"), isAggregate = true)(
     "Return a discrete median of a given set of numbers"
   )
+  val RegrIntercept = mf("regr_intercept", FunctionName("regr_intercept"), Seq(SoQLNumber, SoQLNumber), Seq.empty, SoQLNumber, isAggregate = true) (
+    "Return the OLS Regression intercept of (y, x) columns"
```
Comment (chitang) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:286]: Expand acronyms OLS and R

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala [hunk 2]
```diff
@@ -363,6 +372,9 @@ object SoQLFunctions {
     "Find the remainder(modulus) of one number divided by another"
   )
 
+  val NaturalLog = f("ln", FunctionName("ln"), Map("a" -> NumLike), Seq(VariableType("a")), Seq.empty, VariableType("a"))(
```
Comment (chitang) [inline soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:375]: Shall we add Log base 10 too?  I'll leave that up to you.

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala (hunk 1 above)
Comment (PeterAustinMoore) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala]: 👍  Expanded the acronyms (though r-squared is isn't an acronym, I clarified what it actually is)

File: soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala (hunk 2 above)
Comment (PeterAustinMoore) [reply soql-stdlib/src/main/scala/com/socrata/soql/functions/SoQLFunctions.scala:375]: For our purposes, we just need the `ln` function, but I have a few other SoQL function ideas on deck that I can add a `log(x)` function to as well!

**Review by chitang — APPROVED** [review]

**Pull request merged**